tests/system/src/test/java/test/robot/javafx/scene/DatePickerUpdateOnAlertCloseTest.java

```diff
@@ -0,0 +1,183 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import test.util.Util;
+
+/*
+ * Test for verifying DatePicker update on closing the Alert dialog.
+ *
+ * There is 1 test in this file.
+ * Steps for testDatePickerUpdateOnAlertClose()
+ * 1. Create a alert dialog and add date picker to it.
+ * 2. Add button to scene and show alert dialog on button click.
+ * 3. Click on date picker and select a date from popup.
+ * 4. Verify that selected date is updated in the date picker.
+ */
+
+public class DatePickerUpdateOnAlertCloseTest {
+    static CountDownLatch startupLatch = new CountDownLatch(1);
+    static CountDownLatch onDatePickerShownLatch = new CountDownLatch(1);
+    static CountDownLatch onAlertShownLatch = new CountDownLatch(1);
+    static Robot robot;
+
+    static volatile Stage stage;
+    static volatile Scene scene;
+    static Button button;
+
+    static final int SCENE_WIDTH = 250;
+    static final int SCENE_HEIGHT = SCENE_WIDTH;
+    static final int Y_FACTOR = 5;
+
+    DatePicker datePicker;
+    Alert dialog;
+
+    private void mouseClick(double x, double y) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) (scene.getWindow().getX() + scene.getX() + x),
+                                (int) (scene.getWindow().getY() + scene.getY() + y));
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+    }
+
+    private void mouseClickOnAlertDialog(double x, double y) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) (dialog.getX() + x), (int) (dialog.getY() + y));
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+    }
+
+    private void selectDatePicker() throws Exception {
+        mouseClickOnAlertDialog(datePicker.getLayoutX() + datePicker.getWidth() - 15,
+                                    datePicker.getLayoutY() + datePicker.getHeight() / 2);
+        Thread.sleep(400); // Wait for DatePicker popup to display.
+        Util.waitForLatch(onDatePickerShownLatch, 5, "Failed to show DatePicker popup.");
+    }
+
+    private void showAlertDialog() throws Exception {
+        mouseClick(button.getLayoutX() + button.getWidth() / 2,
+                    button.getLayoutY() + button.getHeight() / 2);
+        Thread.sleep(400); // Wait for Alert dialog to display.
+        Util.waitForLatch(onAlertShownLatch, 5, "Failed to show Alert dialog.");
+    }
+
+    @Test
+    public void testDatePickerUpdateOnAlertClose() throws Exception {
+        Thread.sleep(1000); // Wait for stage to layout
+
+        showAlertDialog();
+        selectDatePicker();
+
+        // Select date from date picker popup
+        mouseClick(datePicker.getLayoutX() + datePicker.getWidth() / 2,
+                    datePicker.getLayoutY() + datePicker.getHeight() * Y_FACTOR);
+
+        Assert.assertFalse(LocalDate.now().isEqual(datePicker.getValue()));
```

Comment:
Do you need a small sleep here? It works for me without it on Windows, but I haven't tried it on my slow Ubuntu 20.04 VM.

Reply:
Even I checked on Windows machine itself which is quite good in configuration. Added small delay of 400ms similar to the delay present where we wait for date picker popup.